### DISCUSSION TITLE
Importing orientations file

### DISCUSSION THREAD

**Discussion by bma698** [mtex-toolbox/mtex#2180] [Ask Anything] [answered]
How I can import orientation file *.txt through the GUI of Mtex? 
![GUI Mtex](https://github.com/user-attachments/assets/82a05755-a686-4c62-b79e-feffc51c83d5)
[Cu_h1_3um_random_modified_data.txt](https://github.com/user-attachments/files/16416961/Cu_h1_3um_random_modified_data.txt)

**Comment by lfgmorales** [comment]
what does your text file contain?

if euler angles, just use the option “data” and properly define which column means

cheers

Luiz


Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.***



On 29 Jul 2024, at 14:05, bma698 ***@***.***> wrote:



How I can import orientation file *.txt through the GUI of Mtex?
GUI.Mtex.jpg (view on web)<https://github.com/user-attachments/assets/82a05755-a686-4c62-b79e-feffc51c83d5>

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2180>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN25I3V62AKHSXPHUODZOYVXBAVCNFSM6AAAAABLUEF7L6VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWHE4TANZQGM>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2180", "url": "https://github.com/mtex-toolbox/mtex/discussions/2180", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by bma698** [reply]
the file contains Euler angles. But when I import it from "data" option, there is no compatible angle convention to define phi1, PHI, and phi2 
![Uploading image.png…]()

**Comment by lfgmorales** [comment]
can you send the data somewhere?

cheers

Luiz



Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.***



On 29 Jul 2024, at 14:16, bma698 ***@***.***> wrote:



the file contains Euler angles. But when I import it from "data" option, there is no compatible angle convention to define phi1, PHI, and phi2
[Uploading image.png…]

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2180#discussioncomment-10179138>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN73QYX4SAWTQMS455TZOYXCDAVCNFSM6AAAAABLUEF7L6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJXHEYTGOA>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2180#discussioncomment-10179138", "url": "https://github.com/mtex-toolbox/mtex/discussions/2180#discussioncomment-10179138", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by bma698** [reply]
Yes sure, the attached file is for the orientations 
[Uploading Cu_h1_3um_random_modified_data.txt…]()

**Comment by lfgmorales** [comment]
Hi

something like this will do it

cheers

Luiz

cs = {crystalSymmetry('m-3m','mineral','Fe')};
[ori,prop] = loadOrientation_generic('Cu_h1_3um_random_modified_data.txt','ColumnNames',{'Euler1','Euler2','Euler3','prop1’})

**Reply by bma698** [reply]
This command works to load the data no problem. But I need also to use these orientations to plot the pole figures. Usually, the GUI helps me to prepare the script, but for this approach, I have just loaded the data but I'm not really able to use them. What I need is to load the data using the GUI

**Reply by ralfHielscher** [reply, marked as answer]
Here come the script to start with :smile: 
```matlab
% define the crystal symmetry
cs = crystalSymmetry('432','mineral','myMineral')

% import the data
ori = orientation.load('Cu_h1_3um_random_modified_data.txt',cs,'ColumnNames',{'Euler1','Euler2','Euler3'});

% define some crystal directions
h = Miller({1,0,0},{1,1,0},{1,1,1},cs);

% plot pole figures
figure(1)
plotPDF(ori,h)

% compute an ODF
odf = calcDensity(ori)

% plot smooth pole figures
figure(2)
plotPDF(odf,h)
```
I hope this helps,
Ralf.

**Reply by bma698** [reply]
Thanks, it works perfectly

**Comment by lfgmorales** [comment]
What exactly do you want to do?

the variable ori is orientations, from there you can plot anything, calculate density, and so on and so forth. Loading with import_wizard will not tell you what and how to plot

To define what pole figures you want to plot, please go to the documentation, an example ia given below

<https://mtex-toolbox.github.io/EBSDOrientationPlots.html>
Plotting Individual Orientations | MTEX<https://mtex-toolbox.github.io/EBSDOrientationPlots.html>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/EBSDOrientationPlots.html>
        [favicon.png] <https://mtex-toolbox.github.io/EBSDOrientationPlots.html>



Cheers

Luiz
